src/components/Select/Select.tsx

```diff
@@ -0,0 +1,270 @@
+            const filteredOptions = options.filter(
+                (option) => !option.hideOption
+            );
+            const updatedItems: SelectOption[] = filteredOptions.map(
+                ({ hideOption, ...option }) => ({
+                    ...option,
+                    classNames: option.selected ? styles.selectedOption : '',
+                })
+            );
+            return (
+                <Menu
+                    {...menuProps}
+                    items={updatedItems}
+                    onChange={(value) => {
+                        const option = updatedItems.find(
+                            (option) => option.value === value
+                        );
+                        toggleOption(option);
+                    }}
+                />
+            );
+        };
+
+        const getSelectedOptionText = () => {
+            if (showPills()) {
+                return undefined;
+            }
+            const selectedOption = options.find((option) => option.selected);
+            return selectedOption?.text;
+        };
+
+        const selectInputProps: TextInputProps = {
+            placeholder: 'Select',
+            clearable: false,
+            inputWidth: TextInputWidth.fill,
+            iconProps: {
+                path: IconName.mdiMenuDown,
+                rotate: dropdownVisible ? 180 : 0,
+            },
+            onClear: onInputClear,
+            ...textInputProps,
+        };
+
+        return (
+            <div
+                className={componentClasses}
+                style={style}
+                ref={ref}
+                data-test-id={dataTestId}
+            >
+                {showPills() ? getPills() : null}
+                <Dropdown
+                    {...dropdownProps}
+                    onVisibleChange={(isVisible) =>
+                        onDropdownVisibilityChange(isVisible)
+                    }
+                    beforeToggleStateChange={beforeToggleStateChange}
+                    overlay={
+                        isLoading ? spinner : <OptionMenu options={options} />
+                    }
+                    classNames={dropdownWrapperClasses}
+                    dropdownClassNames={dropdownMenuOverlayClasses}
+                >
+                    {filterable ? (
+                        <TextInput
+                            {...selectInputProps}
+                            value={
+                                getSelectedOptionText() && !dropdownVisible
+                                    ? getSelectedOptionText()
+                                    : undefined
+                            }
+                            onChange={onInputChange}
+                            classNames={styles.selectInputFilterable}
+                        />
+                    ) : (
+                        <TextInput
+                            {...selectInputProps}
+                            readonly={true}
```

Comment:
The width and size props may be passed via `selectInputProps`, but might be worth considering adding them to the `SelectProps` for visibility.